src/main/java/seedu/duke/models/schema/ModuleWeekly.java

```diff
@@ -1,37 +1,64 @@
 package seedu.duke.models.schema;
 
 
+import org.json.simple.JSONArray;
+import seedu.duke.exceptions.InvalidModuleCodeException;
+import seedu.duke.models.logic.Api;
+
 import java.util.ArrayList;
 
 public class ModuleWeekly extends Module {
+
+    private String moduleCode;
     private int lectureTime;
     private int tutorialTime;
     private int labTime;
     private int lectureDuration;
     private int labDuration;
     private int tutorialDuration;
+
+    private String day;
     private ArrayList<Event> lessons = new ArrayList<Event>();
 
-    public ModuleWeekly(String moduleCode) throws NullPointerException, RuntimeException {
-        super(moduleCode); // Call the constructor of the superclass (Module)
-    }
+
+
 
     public ModuleWeekly(String moduleCode, int lectureTime, int tutorialTime,
-                        int labTime, int lectureDuration, int labDuration, int tutorialDuration)
-            throws NullPointerException, RuntimeException {
+                        int labTime) throws NullPointerException, RuntimeException {
         super(moduleCode);
         this.lectureTime = lectureTime;
         this.tutorialTime = tutorialTime;
         this.labTime = labTime;
-        this.lectureDuration = lectureDuration;
-        this.labDuration = labDuration;
-        this.tutorialDuration = tutorialDuration;
+        //getDuration(moduleCode);
+    }
+
+    public ModuleWeekly(String moduleCode) {
+        super(moduleCode);
+        this.moduleCode = moduleCode;
+        //getDuration(moduleCode);
+        this.lectureTime = 8;
+        this.labTime = 7;
+        this.lectureDuration = 1;
+        this.tutorialTime = 1;
+        this.labDuration = 1;
```

Comment:
I think it would be a good idea to avoid magic numbers. You can do this instead!
<img width="516" alt="image" src="https://github.com/AY2324S1-CS2113-T17-4/tp/assets/110933149/da237ab5-3929-4a61-afba-6c203f32b36c">